File: 백준/Silver/1010. 다리 놓기/다리 놓기.py
# 팩토리얼을 계산하는 함수
def factorial(num):
    result = 1
    for i in range(2, num + 1):
        result *= i
    return result

# 조합을 계산하는 함수
def combination(m, n):
    return factorial(m) // (factorial(n) * factorial(m - n))

# 입력 처리
t = int(input())  # 테스트 케이스 수
result = []

for _ in range(t):
    n, m = map(int, input().split())
    result.append(combination(m, n))  # 조합 계산 결과를 저장

# 결과 출력
for i in result:
    print(i)

File: 백준/Silver/1010. 다리 놓기/다리 놓기.java
import java.io.*;
import java.math.BigInteger;
import java.util.StringTokenizer;

public class Main {
    // 팩토리얼을 계산하는 함수
    static BigInteger factorial(int num) {
        BigInteger result = BigInteger.ONE;
        for (int i = 2; i <= num; i++) {
            result = result.multiply(BigInteger.valueOf(i));
        }
        return result;
    }
    
    // 조합을 계산하는 함수: combination(m, n) = mCn = m! / (n! * (m-n)!)
    static BigInteger combination(int m, int n) {
        return factorial(m).divide(factorial(n).multiply(factorial(m - n)));
    }
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringBuilder sb = new StringBuilder();
        
        int t = Integer.parseInt(br.readLine());
        for (int i = 0; i < t; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int n = Integer.parseInt(st.nextToken());
            int m = Integer.parseInt(st.nextToken());
            sb.append(combination(m, n)).append("\n");
        }
        
        bw.write(sb.toString());
        bw.flush();
        bw.close();
        br.close();
    }
}